### PR TITLE
WorkspaceOwnerSpec: Attribute Tests

### PR DESCRIPTION
**Pull request by marctalbott**
[QA-484](https://broadworkbench.atlassian.net/browse/QA-484)
Tests that check a workspace owner's ability to update the attributes on a workspace. This PR is dependent on [this Workbench-Libs PR](https://github.com/broadinstitute/workbench-libs/pull/219)

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [x] **Submitter**: Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [x] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] Tell your tech lead (TL) that the PR exists if they want to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala [hunk 1]
```diff
@@ -193,6 +196,106 @@ class WorkspaceApiSpec extends TestKit(ActorSystem("MySpec")) with FreeSpecLike
...
+            Rawls.workspaces.updateAttributes(projectName, workspaceName, attributeUpdates)
+            eventually {
+              workspaceResponse(Rawls.workspaces.getWorkspaceDetails(projectName, workspaceName)).workspace.attributes should be (attributeMap)
+            }
+          }
+        }
+      }
+
+      "to delete workspace attributes" - {
+        "from the top" in {
```
Comment (ansingh7115) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala]: I believe deleting ws attributes from the "top" "middle" and "bottom" of the table is testing the _ui table_ to make sure that it updates properly. For the rawls API tests, we just need to test a simple deletion, so top, middle and bottom don't matter. So, I would reduce the "delete workspace attributes" to just one test.

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala [hunk 2]
```diff
@@ -193,6 +196,106 @@ class WorkspaceApiSpec extends TestKit(ActorSystem("MySpec")) with FreeSpecLike
...
+
+              Rawls.workspaces.updateAttributes(projectName, workspaceName, deleteBottomAttribute)
+              eventually {
+                workspaceResponse(Rawls.workspaces.getWorkspaceDetails(projectName, workspaceName)).workspace.attributes should be (attributeMap.filter(attribute => attribute._1.name != "C-key"))
+              }
+            }
+          }
+        }
+
+        "after adding them" in {
```
Comment (ansingh7115) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala]: same with this one - it's testing the ui table to make sure the ui facet loads correctly.

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala (hunk 1 above)
Comment (marctalbott) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala]: Okay great, that will make the tests much simpler

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala [hunk 3]
```diff
@@ -29,7 +32,7 @@ class WorkspaceApiSpec extends TestKit(ActorSystem("MySpec")) with FreeSpecLike
   val owner: Credentials = UserPool.chooseProjectOwner
   val ownerAuthToken: AuthToken = owner.makeAuthToken()
 
-  implicit override val patienceConfig = PatienceConfig(timeout = scaled(Span(5, Minutes)), interval = scaled(Span(20, Seconds)))
+  implicit override val patienceConfig: PatienceConfig = PatienceConfig(timeout = scaled(Span(5, Minutes)), interval = scaled(Span(20, Seconds)))
```
Comment (aweng98) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala]: the firecloud-ui test has patienceConfig set at 10 seconds. 5 min patienceConfig is long time. test should fail if this operation takes more than 10 seconds. poor performance is a red flag should be investigated.

**Comment by aweng98** [comment]
well-written test.

**Review by aweng98 — APPROVED** [review]

**Review by ansingh7115 — APPROVED** [review]

**Pull request merged**